File: src/Main/java/com/nishant/algorithms/Sorting/SelectionSort/SelectionSort.java
package com.nishant.algorithms.Sorting.SelectionSort;

public class SelectionSort {

    public static void sort(int[] arr) {
        for(int start = 0; start < arr.length - 1; start ++) {
            int min = Integer.MAX_VALUE, minIndex = -1;

            for(int i = start; i < arr.length; i ++) {
                if(arr[i] < min) {
                    min = arr[i];
                    minIndex = i;
                }
            }

            swap(arr, start, minIndex);
        }
    }

    private static void swap(int[] arr, int pos1, int pos2) {
        int temp = arr[pos1];
        arr[pos1] = arr[pos2];
        arr[pos2] = temp;
    }

}

File: src/main/python/com/nishant/algorithms/sorting/selectionsort/selection_sort.py
import math


def selection_sort(to_sort):
    for start in range(len(to_sort)):
        min_num = math.inf
        min_index = -1

        for j in range(start, len(to_sort), 1):
            if to_sort[j] < min_num:
                min_num = to_sort[j]
                min_index = j

        to_sort[start], to_sort[min_index] = to_sort[min_index], to_sort[start]


if __name__ == "__main__":
    arr = [1, 2, 6, 4, 5]
    print(str(arr))
    selection_sort(arr)
    print(str(arr))
